metrics/performance/first-meaningful-paint.js

```diff
@@ -18,49 +18,61 @@
 
 const DevtoolsTimelineModel = require('devtools-timeline-model');
 
+const FAILURE_ERROR = new Error('Navigation and first paint timings not found');
+
 class FirstMeaningfulPaint {
+
   /**
    * @param {!Array<!Object>} traceData
    */
   static parse(traceData) {
     return new Promise((resolve, reject) => {
+      if (!traceData || !Array.isArray(traceData)) {
+        return reject(FAILURE_ERROR);
+      }
+
       const model = new DevtoolsTimelineModel(traceData);
       const events = model.timelineModel().mainThreadEvents();
-      const ret = {};
 
-      const navStartEvent = events
-          .filter(FirstMeaningfulPaint._filterEventsForNavStart);
-      const firstTextPaintEvent = events
-          .filter(FirstMeaningfulPaint._filterEventsForFirstTextPaint);
+      // Identify the frameID of the main frame
```

Comment:
maybe specify the main frame is the last one? (if I'm understanding what you're doing here correctly)